app/themes/northernireland/templates/introduction.html

```diff
@@ -51,7 +51,7 @@ <h2 class="saturn">You are completing this for
             {% endif %}
 
             {%- if legal_basis -%}
-                <h2 class="neptune u-mb-xs">Your response is legally required</h2>
+                <h2 class="neptune u-mb-xs" id="legal-response">Your response is legally required</h2>
                 <p class="mars u-mb-m">{{ legal_basis }}</p>
```

Comment:
I would add an 'id' to `legal_basis` and then test the content of that in the functional test. This is the variable context we're actually trying to test here.